mentorship ios/ViewModels/RelationViewModel.swift

```diff
@@ -0,0 +1,162 @@
+            .receive(on: RunLoop.main)
+            .catch { _ in Just(self.currentRelation) }
+            .sink { [weak self] current in
+                //use current relation data
+                self?.currentRelation = current
+                self?.personName = self?.getPersonNameAndType(data: current) ?? ""
+                self?.inActivity = false
+                //chain api call. get current tasks using id from current relation
+                self?.fetchTasks(id: current.id ?? 0, token: token)
+        }
+    }
+    
+    func fetchTasks(id: Int, token: String) {
+        self.tasksCancellable = NetworkManager.callAPI(urlString: URLStringConstants.MentorshipRelation.getCurrentTasks(id: id), token: token)
+            .receive(on: RunLoop.main)
+            .catch { _ in Just(self.tasks) }
+            .sink { [weak self] tasks in
+                //if success
+                if NetworkManager.responseCode == 200 {
+                    self?.doneTasks.removeAll()
+                    self?.toDoTasks.removeAll()
+                    //assign done tasks and tasks to do
+                    for task in tasks {
+                        if task.isDone ?? false {
+                            self?.doneTasks.append(task)
+                        } else {
+                            self?.toDoTasks.append(task)
+                        }
+                    }
+                }
+                //else show fail alert
+                else {
+                    self?.showErrorAlert = true
+                    self?.alertMessage = LocalizableStringConstants.operationFail
+                }
+        }
+    }
+    
+    //func to get name of other person in current relation.
+    func getPersonNameAndType(data: HomeModel.HomeResponseData.RequestStructure) -> String {
+        //get user profile
+        let userProfile = ProfileViewModel().getProfile()
+        //match users name with mentee name.
+        //if different, return mentee's name. Else, return mentor's name
+        //Logic: Person with different name is in relation with us.
+        if data.mentee?.name != userProfile.name {
+            self.personType = LocalizableStringConstants.mentee
+            return data.mentee?.name ?? ""
+        } else {
+            self.personType = LocalizableStringConstants.mentor
+            return data.mentor?.name ?? ""
+        }
+    }
+    
+    //mark task as complete api call + data change
+    func markAsComplete() {
+        //get auth token
+        guard let token = try? KeychainManager.getToken() else {
+            return
+        }
+        
+        //get tapped task
+        let taskTapped = RelationViewModel.taskTapped
+        
+        //api call
+        cancellable = NetworkManager.callAPI(
+            urlString: URLStringConstants.MentorshipRelation.markAsComplete(reqID: currentRelation.id ?? 0, taskID: taskTapped.id ?? 0),
```

Comment:
it is not a good practice to pass random currentRelation.id and taskTapped.id i.e. 0.
It may create a major bug in the app. rather than using nil coalescing, use guard or if let statements. 
Sorry to put this now, But from now onwards please keep that in mind.
I have seen this in many places, please try to solve them in refactor PR.
Example approach: 
<img width="1009" alt="Screenshot 2020-07-06 at 11 34 44 PM" src="https://user-images.githubusercontent.com/30840527/86624645-47b7dc00-bfe1-11ea-99cf-dbddceef6174.png">

Reply:
okay